### PR TITLE
CORE-6720 - When same role is being assigned to the same user, getting incorrect response status code

### PR DESCRIPTION
**Pull request by pmck91**

### CODE REVIEW

**Review by conal-smith — CHANGES_REQUESTED** [review]
Let's not bring dependency on http-rpc into this `permission-common` (and thereby `permission-storage-writer-impl` as this runs on the storage worker, it doesn't know about http.

Instead, have a look at `PermissionManagementHandler` to see how the mapping is performed. This is where responses from the storage worker are translated to HTTP specific responses.

You could change the mapping:
```
EntityAssociationAlreadyExistsException::class.java.name -> throw InvalidInputDataException(e.message!!)
```
To throw `ResourceAlreadyExistsException`, which has the CONFLICT response code.

**Comment by pmck91** [comment]
> To throw ResourceAlreadyExistsException, which has the CONFLICT response code.

Oh nice, much easier, I'll update that now

**Review by conal-smith — DISMISSED** [review]
LGTM, you'll need to update some tests that expected the old HTTP status code.

**Comment by vkolomeyko** [comment]
CI build reports E2E tests failures like so:
https://ci02.dev.r3.com/blue/organizations/jenkins/Corda5%2Fcorda-runtime-os-e2e-tests/detail/PR-2158/3/tests
```
java.lang.AssertionError: 
Expecting:
  <net.corda.httprpc.client.exceptions.InternalErrorException: "{\n    \"title\": \"Permission '2ff032a2-a946-42b8-93d0-98b6981b77ba' is already associated with Role 'd7a7485b-7ef3-41e9-8d4e-e4a3bdf647f4'.\",\n    \"status\": 409,\n    \"type\": \"https://javalin.io/documentation#error-responses\",\n    \"details\": {\"code\":\"CONFLICT\"}\n}">
to be an instance of:
  <net.corda.httprpc.client.exceptions.RequestErrorException>
but was:
  <"net.corda.httprpc.client.exceptions.InternalErrorException: "{\n    \"title\": \"Permission '2ff032a2-a946-42b8-93d0-98b6981b77ba' is already associated with Role 'd7a7485b-7ef3-41e9-8d4e-e4a3bdf647f4'.\",\n    \"status\": 409,\n    \"type\": \"https://javalin.io/documentation#error-responses\",\n    \"details\": {\"code\":\"CONFLICT\"}\n}"
	at net.corda.httprpc.client.connect.remote.RemoteUnirestClient.doCall(RemoteClient.kt:111)
	at net.corda.httprpc.client.connect.remote.RemoteUnirestClient.call(RemoteClient.kt:54)
	at net.corda.httprpc.client.connect.HttpRpcClientProxyHandler.invoke(HttpRpcClientProxyHandler.kt:115)
	at com.sun.proxy.$Proxy40.addPermission(Unknown Source)
	at net.corda.applications.workers.rpc.CreateRoleE2eTest.test_add_and_remove_permission_to_a_role$lambda-8$lambda-7(CreateRoleE2eTest.kt:138)
	at org.assertj.core.api.ThrowableAssert.catchThrowable(ThrowableAssert.java:62)
	at org.assertj.core.api.AssertionsForClassTypes.catchThrowable(AssertionsForClassTypes.java:739)
	at org.assertj.core.api.Assertions.catchThrowable(Assertions.java:1163)
	at org.assertj.core.api.Assertions.assertThatThrownBy(Assertions.java:1032)
	at net.corda.applications.workers.rpc.CreateRoleE2eTest.test add and remove permission to a role(CreateRoleE2eTest.kt:138)
```
FAO: @pmck91

File: libs/http-rpc/http-rpc-client/src/main/kotlin/net/corda/httprpc/client/connect/remote/RemoteClient.kt [hunk 1]
```diff
@@ -107,6 +108,7 @@ internal class RemoteUnirestClient(override val baseAddress: String, private val
                     HttpStatus.FORBIDDEN, HttpStatus.UNAUTHORIZED, HttpStatus.METHOD_NOT_ALLOWED, HttpStatus.PROXY_AUTHENTICATION_REQUIRED
                     -> throw PermissionException(errorResponseJson)
                     HttpStatus.NOT_FOUND -> throw MissingRequestedResourceException(errorResponseJson)
+                    HttpStatus.CONFLICT -> throw ResourceAlreadyExistsException (errorResponseJson)
```
Comment (vkolomeyko) [inline libs/http-rpc/http-rpc-client/src/main/kotlin/net/corda/httprpc/client/connect/remote/RemoteClient.kt]: Minor:
```suggestion
                    HttpStatus.CONFLICT -> throw ResourceAlreadyExistsException(errorResponseJson)
```

**Review by vkolomeyko — DISMISSED** [review]
LGTM

One minor comment posted. If there will be a need for further changes - please address it. Otherwise, I am happy to merge "as is".

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Pull request merged**